### PR TITLE
feat: Port tests from `tap` to `node:assert` and `node:test`

### LINKED ISSUE #140: Updating for v9
Hi, I would like to migrate the tests from tap to `node:test`

### PR DESCRIPTION
**Pull request by codershiba**
Closes #140

Ports tests from tap to `node:assert`, `node:test`, `borp`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   "scripts": {
     "lint": "standard | snazzy",
     "lint-ci": "standard",
-    "test": "tap --no-cov",
-    "test-ci": "tap --cov --no-check-coverage --coverage-report=text",
+    "test": "borp -p \"test/**/*.js\"",
+    "test-ci": "borp --coverage -p \"test/**/*.js\"",
```
Comment (jsumners) [inline package.json:12]: Utilizing single quotes will prevent shell expansion and provide the glob directly to the runner tool:

```suggestion
    "test": "borp -p 'test/**/*.js'",
    "test-ci": "borp --coverage -p 'test/**/*.js'",
```

@mcollina does your tool support platform agnostic glob expansion?

File: test/res.test.js [hunk 2]
```diff
@@ -2,63 +2,58 @@
...
+    assert.ok(serialized.raw)
+    assert.strictEqual(serialized.raw.statusCode, 200)
     res.end()
   }
 })
 
-test('can wrap response serializers', function (t) {
-  t.plan(3)
-
+test('can wrap response serializers', () => {
```
Comment (mcollina) [inline test/res.test.js:45]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/res.test.js [hunk 3]
```diff
@@ -67,43 +62,39 @@ test('can wrap response serializers', function (t) {
...
     const serialized = serializer(res)
-    t.notOk(serialized.statusCode)
+    assert.ok(!serialized.statusCode)
   }
 })
 
-test('res.headers is serialized', function (t) {
-  t.plan(1)
-
+test('res.headers is serialized', () => {
```
Comment (mcollina) [inline test/res.test.js:69]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/res.test.js [hunk 4]
```diff
@@ -67,43 +62,39 @@ test('can wrap response serializers', function (t) {
...
-    t.equal(serialized.headers['x-custom'], 'y')
+    assert.strictEqual(serialized.headers['x-custom'], 'y')
     res.end()
   }
 })
 
-test('res.statusCode is null before headers are flushed', function (t) {
-  t.plan(1)
-
+test('req.url will be obtained from input request url when input request url is not an object', () => {
```
Comment (mcollina) [inline test/res.test.js:86]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/res.test.js [hunk 5]
```diff
@@ -2,63 +2,58 @@
...
-    t.equal(serialized.propertyIsEnumerable('raw'), false)
+    assert.strictEqual(serialized.propertyIsEnumerable('raw'), false)
     res.end()
   }
 })
 
-test('res.raw is available', function (t) {
-  t.plan(2)
-
+test('res.raw is available', () => {
```
Comment (mcollina) [inline test/res.test.js:27]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/res.test.js [hunk 6]
```diff
@@ -2,63 +2,58 @@
...
+const http = require('node:http')
+const { after, test } = require('node:test')
 const serializers = require('../lib/res')
-const wrapResponseSerializer = require('../').wrapResponseSerializer
-
-test('res.raw is not enumerable', function (t) {
-  t.plan(1)
+const { wrapResponseSerializer } = require('../')
 
+test('res.raw is not enumerable', () => {
```
Comment (mcollina) [inline test/res.test.js:11]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 7]
```diff
@@ -1,433 +1,389 @@
...
+const http = require('node:http')
+const { after, test } = require('node:test')
 const serializers = require('../lib/req')
-const wrapRequestSerializer = require('../').wrapRequestSerializer
-
-test('maps request', function (t) {
-  t.plan(2)
+const { wrapRequestSerializer } = require('../')
 
+test('maps request', () => {
```
Comment (mcollina) [inline test/req.test.js:9]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 8]
```diff
@@ -1,433 +1,389 @@
...
+    assert.ok(serialized.req)
+    assert.ok(serialized.req.method)
     res.end()
   }
 })
 
-test('does not return excessively long object', function (t) {
-  t.plan(1)
-
+test('does not return excessively long object', () => {
```
Comment (mcollina) [inline test/req.test.js:26]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 9]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(Object.keys(serialized).length, 6)
+    assert.strictEqual(Object.keys(serialized).length, 6)
     res.end()
   }
 })
 
-test('req.raw is available', function (t) {
-  t.plan(2)
-
+test('req.raw is available', () => {
```
Comment (mcollina) [inline test/req.test.js:42]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 10]
```diff
@@ -1,433 +1,389 @@
...
+    assert.ok(serialized.raw)
+    assert.strictEqual(serialized.raw.foo, 'foo')
     res.end()
   }
 })
 
-test('req.raw will be obtained in from input request raw property if input request raw property is truthy', function (t) {
-  t.plan(2)
-
+test('req.raw will be obtained in from input request raw property if input request raw property is truthy', () => {
```
Comment (mcollina) [inline test/req.test.js:60]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 11]
```diff
@@ -1,433 +1,389 @@
...
+    assert.ok(serialized.raw)
+    assert.strictEqual(serialized.raw.req.foo, 'foo')
     res.end()
   }
 })
 
-test('req.id defaults to undefined', function (t) {
-  t.plan(1)
-
+test('req.id defaults to undefined', () => {
```
Comment (mcollina) [inline test/req.test.js:78]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 12]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(serialized.id, undefined)
+    assert.strictEqual(serialized.id, undefined)
     res.end()
   }
 })
 
-test('req.id has a non-function value', function (t) {
-  t.plan(1)
-
+test('req.id has a non-function value', () => {
```
Comment (mcollina) [inline test/req.test.js:94]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 13]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(typeof serialized.id === 'function', false)
+    assert.strictEqual(typeof serialized.id === 'function', false)
     res.end()
   }
 })
 
-test('req.id will be obtained from input request info.id when input request id does not exist', function (t) {
-  t.plan(1)
-
+test('req.id will be obtained from input request info.id when input request id does not exist', () => {
```
Comment (mcollina) [inline test/req.test.js:110]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 14]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(serialized.id, 'test')
+    assert.strictEqual(serialized.id, 'test')
     res.end()
   }
 })
 
-test('req.id has a non-function value with custom id function', function (t) {
-  t.plan(2)
-
+test('req.id has a non-function value with custom id function', () => {
```
Comment (mcollina) [inline test/req.test.js:127]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 15]
```diff
@@ -1,433 +1,389 @@
...
+    assert.strictEqual(typeof serialized.id === 'function', false)
+    assert.strictEqual(serialized.id, 42)
     res.end()
   }
 })
 
-test('req.url will be obtained from input request req.path when input request url is an object', function (t) {
-  t.plan(1)
-
+test('req.url will be obtained from input request req.path when input request url is an object', () => {
```
Comment (mcollina) [inline test/req.test.js:145]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 16]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(serialized.url, '/test')
+    assert.strictEqual(serialized.url, '/test')
     res.end()
   }
 })
 
-test('req.url will be obtained from input request url.path when input request url is an object', function (t) {
-  t.plan(1)
-
+test('req.url will be obtained from input request url.path when input request url is an object', () => {
```
Comment (mcollina) [inline test/req.test.js:162]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 17]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(serialized.url, '/test')
+    assert.strictEqual(serialized.url, '/test')
     res.end()
   }
 })
 
-test('req.url will be obtained from input request url when input request url is not an object', function (t) {
-  t.plan(1)
-
+test('req.url will be obtained from input request url when input request url is not an object', () => {
```
Comment (mcollina) [inline test/req.test.js:179]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 18]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(serialized.url, '/test')
+    assert.strictEqual(serialized.url, '/test')
     res.end()
   }
 })
 
-test('req.url will be empty when input request path and url are not defined', function (t) {
-  t.plan(1)
-
+test('req.url will be empty when input request path and url are not defined', () => {
```
Comment (mcollina) [inline test/req.test.js:196]: You need the plan here. Use https://github.com/mcollina/tspl

File: test/req.test.js [hunk 19]
```diff
@@ -1,433 +1,389 @@
...
-    t.equal(serialized.url, '/')
+    assert.strictEqual(serialized.url, '/')
     res.end()
   }
 })
 
-test('req.url will be obtained from input request originalUrl when available', function (t) {
-  t.plan(1)
-
+test('req.url will be obtained from input request originalUrl when available', () => {
```
Comment (mcollina) [inline test/req.test.js:212]: You need the plan here. Use https://github.com/mcollina/tspl

**Review by mcollina — CHANGES_REQUESTED** [review]
If the test is asynchronous, you need to use an async function and add a test plan with https://github.com/mcollina/tspl.

**Comment by codershiba** [comment]
Hi @mcollina , can you take a look here?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Can you drop node v14 and v16 from  CI?

File: .github/workflows/ci.yml [hunk 20]
```diff
@@ -39,7 +39,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node-version: [14, 16, 18]
+        node-version: [18]
```
Comment (mcollina) [inline .github/workflows/ci.yml:42]:
```suggestion
        node-version: [18, 20]
```

**Comment by jsumners** [comment]
Please resolve the conflicts.

**Comment by codershiba** [comment]
> Please resolve the conflicts.

Resolved

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by codershiba** [comment]
@mcollina is there anything else left to do here?

**Pull request merged**